lldp_util.c

```diff
@@ -706,10 +704,23 @@ int is_macvtap(const char *ifname)
 		goto out_free;
 	}
 
-	memset(nlh, 0, NLMSG_SIZE);
+	do {
+		realsize = recv(s, NULL, 0, MSG_DONTWAIT | MSG_PEEK | MSG_TRUNC);
+	} while ((realsize < 0) && errno == EINTR);
+
+	if (realsize <= 0) {
```

Comment:
This should probably be `< 0`, otherwise it will never consume a zero-sized packet which is legitimate for some protocols like UDP, although I can't imagine netlink using that for anything. However, it is probably better to attempt to consume it if we are being pedantic here.